### ISSUE TITLE
ERROR com.github.ccob.bittrex4j.BittrexExchange - Failed to parse response

### ISSUE THREAD

**Issue by szd116** [CCob/bittrex4j#42]
Hello everyone,

I have been getting the following messages for the past few weeks. I am running the latest jar in production, any ideas why the exceptions ? thanks. 

2018-09-18 16:43:00:217 -0400 [Thread-14746] ERROR com.github.ccob.bittrex4j.BittrexExchange - Failed to parse response
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "FI" (class com.github.ccob.bittrex4j.dao.Fill), not marked as ignorable (15 known properties: "Rate", "Total", "OrderType", "Price", "Quantity", "TimeStamp", "Id", "FillType", "total", "orderType", "quantity", "timeStamp", "id", "fillType", "price"])
 at [Source: (org.apache.http.client.entity.DeflateInputStream); line: 1, column: 204] (through reference chain: com.github.ccob.bittrex4j.dao.UpdateExchangeState["f"]->java.lang.Object[][0]->com.github.ccob.bittrex4j.dao.Fill["FI"])
        at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
        at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:822)
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1152)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1582)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperties(BeanDeserializerBase.java:1536)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:504)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1280)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
        at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:195)
        at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:21)
        at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:529)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:528)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:417)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1280)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
        at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:1611)
        at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1188)
        at com.github.ccob.bittrex4j.BittrexExchange.lambda$registerForEvent$1(BittrexExchange.java:196)
        at com.github.signalr4j.client.hubs.HubProxy$13.run(HubProxy.java:499)
        at jdk.internal.reflect.GeneratedMethodAccessor10.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at com.github.signalr4j.client.hubs.HubProxy$8.run(HubProxy.java:446)
        at com.github.signalr4j.client.hubs.HubProxy$8.run(HubProxy.java:423)
        at com.github.signalr4j.client.hubs.Subscription.onReceived(Subscription.java:31)
        at com.github.signalr4j.client.hubs.HubProxy.invokeEvent(HubProxy.java:413)
        at com.github.signalr4j.client.hubs.HubConnection.onReceived(HubConnection.java:118)
        at com.github.signalr4j.client.transport.TransportHelper.processReceivedData(TransportHelper.java:96)
        at com.github.signalr4j.client.Connection.processReceivedData(Connection.java:755)
        at com.github.signalr4j.client.Connection.access$000(Connection.java:29)
        at com.github.signalr4j.client.Connection$10.onData(Connection.java:693)
        at com.github.signalr4j.client.transport.WebsocketTransport$1.onMessage(WebsocketTransport.java:127)
        at org.java_websocket.client.WebSocketClient.onWebsocketMessage(WebSocketClient.java:360)
        at org.java_websocket.drafts.Draft_6455.processFrame(Draft_6455.java:696)
        at org.java_websocket.WebSocketImpl.decodeFrames(WebSocketImpl.java:367)
        at org.java_websocket.WebSocketImpl.decode(WebSocketImpl.java:212)
        at org.java_websocket.client.WebSocketClient.run(WebSocketClient.java:287)
        at java.base/java.lang.Thread.run(Thread.java:844)

**Comment by szd116** [comment]
It seems like Bittrex had a major system upgrade thus broke many of the api2.0, @CCob would you please take a look ? Thanks.

**Comment by CCob** [comment]
I've just pushed 1.0.11-SNAPSHOT which includes a fix for this.  Let me know if it works for you.  Bittrex have added a new Fill Id to the Fills data structure.

**Comment by CheremisinVlad** [comment]
How can i add this snapshot for my maven project. I can't find it in maven central.

**Comment by CCob** [comment]
`Either rate or price should be set` - this indicates a Fill object has been constructed but neither rate or price has been specified in the constructor.  Have they added another value to indicate what the fill price was I wonder.

**Comment by szd116** [comment]
@CCob Thanks for the help it works. @CheremisinVlad you can go to the home page of this project and click on develop and then choose branch to see the snapshot.

**Comment by revidee** [comment]
> @CCob Thanks for the help it works. @CheremisinVlad you can go to the home page of this project and click on develop and then choose branch to see the snapshot.

How did you solve that issue? Did they add new parameters in the response?
I get a somewhat similar error:
`ERROR [Thread-14] (SignalRLoggerDecorator.java:30) - HubConnection - Error: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `com.github.ccob.bittrex4j.dao.Fill`, problem: null
 at [Source: (org.apache.http.client.entity.DeflateInputStream); line: 1, column: 3485] (through reference chain: com.github.ccob.bittrex4j.dao.UpdateExchangeState["f"]->java.lang.Object[][0])`

**Comment by revidee** [comment]
Additionally, this happens when you call the getMarketSummaries REST method:
 WARN [main] (BittrexExchange.java:552) - Request to URL
 ```
https://bittrex.com/api/v2.0/pub/markets/getmarketsummaries failed with error Unrecognized field "IsRestricted" (class com.github.ccob.bittrex4j.dao.Market), not marked as ignorable (20 known properties: "BaseCurrency", "LogoUrl", "MarketName", "Created", "MarketCurrencyLong", "MarketCurrency", "IsSponsored", "IsActive", "MinTradeSize", "Notice", "BaseCurrencyLong", "baseCurrency", "logoUrl", "marketName", "marketCurrencyLong", "marketCurrency", "minTradeSize", "notice", "baseCurrencyLong", "created"])
 at [Source: (StringReader); line: 1, column: 424] (through reference chain: com.github.ccob.bittrex4j.dao.Response["result"]->java.lang.Object[][0]->com.github.ccob.bittrex4j.dao.MarketSummaryResult["Market"]->com.github.ccob.bittrex4j.dao.Market["IsRestricted"]), retries left: 4
```

seems like a new field, "isRestricted" boolean got added.

**Comment by revidee** [comment]
So to sum up, there are probably multiple new fields in the API that currently break some methods. I will continue to make pull-request as I discover these.

Cheers

**Comment by CCob** [comment]
Various pull requests from @revidee (thanks) and one from me should fix most of the issues now.  I have just published 1.0.11 release which should be available on maven central within the hour.
